reference-tests/src/test/resources/templates/BlockchainReferenceTest.java.template

```diff
@@ -49,15 +50,14 @@ public class %%TESTS_NAME%% {
 
   public static Stream<Arguments> getTestParametersForConfig()
         throws ExecutionException, InterruptedException {
-    if (FAILED_MODULE.isEmpty()) {
-    return generateTestParametersForConfig(TEST_CONFIG_FILE_DIR_PATH).stream().map(params ->
-        Arguments.of(params[0], params[1], params[2])
-        );
-    } else {
-      return generateTestParametersForConfig(TEST_CONFIG_FILE_DIR_PATH, FAILED_MODULE, FAILED_CONSTRAINT).stream().map(params ->
-          Arguments.of(params[0], params[1], params[2])
-      );
+Collection<Object[]> testParameters = FAILED_MODULE.isEmpty()
+        ? generateTestParametersForConfig(TEST_CONFIG_FILE_DIR_PATH)
+        : generateTestParametersForConfig(TEST_CONFIG_FILE_DIR_PATH, FAILED_MODULE, FAILED_CONSTRAINT);
+    Collection<Object[]> filteredTestParameters = testParameters.stream().filter(params -> (Boolean) params[2]).toList();
```

Comment:
This ensures that the test configs that were going to be skipped will be filtered here itself resulting in a cleaner test report without 10s of thousands of skipped tests. I still need to ensure if all test configs are filtered there is atleast one dummy test config returned. This is because junit expects atleast 1 set of parameters for parameterized test.